README.md

```diff
@@ -1,67 +1,123 @@
+
+
+## Using Flow
+
+Gradle:
+
+```groovy
+compile 'com.squareup.flow:flow:1.0-alpha'
 ```
 
-Flow keeps track of your state history, so you can go back.
+Install Flow into your Activity:
 
 ```java
-flow.goBack();
+public class MainActivity {
+  @Override protected void attachBaseContext(Context baseContext) {
+    baseContext = Flow.configure(baseContext, this).install();
+    super.attachBaseContext(baseContext);
+  }
+}
 ```
 
-Your app provides Flow with a Dispatcher which executes state changes.
+By default, Flow will take over your Activity's content view. When you start your Activity, you should see a "Hello world" screen. Of course you'll want to change this-- that's covered under [Controlling UI](#Controlling-UI) below.
+
+### Defining UI states with key objects
+
+Your Activity's UI states are represented in Flow by Objects, which Flow refers to as "keys". Keys are typically [value objects][valueobject] with just enough information to identify a discrete UI state.
+
+Flow relies on a key's [equals][equals] and [hashCode][hashcode] methods for its identity. Keys should be immutable-- that is, their `equals` and `hashCode` methods should always behave the same.
+
+To give an idea of what keys might look like, here are some examples:
 
 ```java
-flow.setDispatcher(new Flow.Dispatcher() {
- @Override public void dispatch(Traversal traversal, TraversalCallback callback) {
-      Object newState = traversal.destination.top();
-      displayViewFor(newState);
-      callback.onTraversalCompleted();
-    }
-});
-```
+public enum TabKey {
+  TIMELINE,
+  NOTIFICATIONS,
+  PROFILE
+}
 
+public final class HomeKey extends flow.ClassKey {
+}
 
-Download
---------
+public final class ArticleKey {
+  public final String articleId;
 
-Download [the latest JAR][1] or grab via Maven:
+  public ArticleKey(String articleId) {
+    this.articleId = articleId;
+  }
 
-```xml
-<dependency>
-    <groupId>com.squareup.flow</groupId>
-    <artifactId>flow</artifactId>
-    <version>(insert latest version)</version>
-</dependency>
-```
-or Gradle:
-```groovy
-compile 'com.squareup.flow:flow:(latest version)'
+  public boolean equals(Object o) {
+    return (o instanceof ArticleKey) &&
+           articleId.equals(((ArticleKey)o).articleId);
+  }
+  
+  public int hashCode() {
+    return articleId.hashCode();
+  }
+}
 ```
 
+See the [Sample Projects](#sample-projects) below for more example keys.
+
+
+### Navigation
+Flow offers simple commands for navigating within your app.
+
+`Flow#goBack()` -- Goes back to the previous [key][keys]. Think "back button".
+
+`Flow#set(key)` -- Goes to the requested key. Goes back or forward depending on whether the key is already in the History.
+
+Flow also lets you rewrite history safely and easily.
+
+`Flow#setHistory(history, direction)` -- Change history to whatever you want.
+
+See the [Flow][Flow.java] class for other convenient operators.
 
+### History
```

Comment:
Maybe just merge it into "Navigation", and retitle that section "Navigation and History"?